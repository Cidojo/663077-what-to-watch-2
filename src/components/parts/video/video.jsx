import * as React from 'react';
import * as PropTypes from 'prop-types';

const Video = React.forwardRef((props, ref) => {
  const {posterImage, videoLink, muted} = props;

  return (
    <video
      className="player__video"
      ref={ref}
      src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
      poster={posterImage}
      muted={muted}
    />
  );
});

Video.propTypes = {
  videoLink: PropTypes.string,
  posterImage: PropTypes.string,
  muted: PropTypes.bool
};

Video.defaultProps = {
  videoLink: ``,
  posterImage: ``,
  muted: false
};

export {Video};
