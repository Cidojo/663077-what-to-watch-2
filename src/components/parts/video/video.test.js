import * as React from 'react';
import {Video} from './video.jsx';
import * as renderer from 'react-test-renderer';

const props = {
  videoLink: ``,
  posterImage: ``,
  muted: false
};

it(`should render Video component without errors`, () => {
  const tree = renderer.create(<Video {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
