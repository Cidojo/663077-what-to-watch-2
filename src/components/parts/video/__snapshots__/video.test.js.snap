// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Video component without errors 1`] = `
<video
  className="player__video"
  muted={false}
  poster=""
  src=""
/>
`;
